services/datamanager/datasync/sync.go

```diff
@@ -56,24 +56,29 @@ type ManagerConstructor func(logger golog.Logger, cfg *config.Config, lastModMil
 
 // NewDefaultManager returns the default Manager that syncs data to app.viam.com.
 func NewDefaultManager(logger golog.Logger, cfg *config.Config, lastModMillis int) (Manager, error) {
-	tlsConfig := config.NewTLSConfig(cfg).Config
-	cloudConfig := cfg.Cloud
-	rpcOpts := []rpc.DialOption{
-		rpc.WithTLSConfig(tlsConfig),
-		rpc.WithEntityCredentials(
-			cloudConfig.ID,
-			rpc.Credentials{
-				Type:    rdkutils.CredentialsTypeRobotSecret,
-				Payload: cloudConfig.Secret,
-			}),
-	}
+	if cfg.Cloud != nil && cfg.Cloud.AppAddress != "" {
+		tlsConfig := config.NewTLSConfig(cfg).Config
+		cloudConfig := cfg.Cloud
+		rpcOpts := []rpc.DialOption{
+			rpc.WithTLSConfig(tlsConfig),
+			rpc.WithEntityCredentials(
+				cloudConfig.ID,
+				rpc.Credentials{
+					Type:    rdkutils.CredentialsTypeRobotSecret,
+					Payload: cloudConfig.Secret,
+				}),
+		}
 
-	conn, err := NewConnection(logger, cfg.Cloud.AppAddress, rpcOpts)
-	if err != nil {
-		return nil, err
+		conn, err := NewConnection(logger, cfg.Cloud.AppAddress, rpcOpts)
+		if err != nil {
+			return nil, err
+		}
+		client := NewClient(conn)
+		return NewManager(logger, cfg.Cloud.ID, client, conn, lastModMillis)
+	} else {
+		logger.Debug("Using no-op sync manager when Cloud config is not available")
```

Comment:
opt: flip the if case here so you can return early and dedent the larger block of logic

Reply:
good suggestion